Configuration/SFA.DAS.Configuration.AzureTableStorage/ConfigurationBootstrapper.cs

```
@@ -0,0 +1,36 @@
+using System;
+using Microsoft.Extensions.Configuration;
+
+namespace SFA.DAS.Configuration.AzureTableStorage
+{
+    public class ConfigurationBootstrapper
+    {
+        private const string DefaultEnvironment = "LOCAL";
+
+        public static (string StorageConnectionString, string EnvironmentName) GetEnvironmentVariables()
+        {
+            var environmentVariablesConfig = new ConfigurationBuilder().AddEnvironmentVariables().Build();
```

What do you think to also including json config? e.g.

```c#
var builder = new ConfigurationBuilder()
    .SetBasePath(Directory.GetCurrentDirectory())
    .AddJsonFile("appsettings.json", true, true)
    .AddJsonFile("appsettings.Development.json", true, true)
    .AddEnvironmentVariables();
```

---

The bootstrapper's job is to do just enough to get the storage connection string and environment name from their environment variables, so for example, they can be passed to `AddAzureTableStorageConfiguration` when creating the host.

The appsettings.json files can be added during host creation as per normal.

---

Ok that makes sense, we should probably default to `UseDevelopmentStorage=true` (similar to `SFA.DAS.AutoConfiguration`) in that case instead of throwing an exception if it can't be found via the environment variable. What do you think?

---

Initially I thought that's a good idea, but when I started to make the change, I thought perhaps we should leave it to throw an exception. My reasoning is...
* defaulting to use the emulator is only applicable in the `LOCAL` developer environment (other environments should throw if the environment variable is missing anyway, as an emulator isn't available).
* at least in our team, every developer is either running Azurite in a container, or using a real cloud storage account. Both options require a real storage connection string to be present in the environment variable anyway.
* if the environment variable is missing, the exception thrown gives the developer the exact information required to remedy the situation.

---

I'd prefer to keep things as simple as possible for a developer to get up and running. Admittedly setting an environment variable, whether via the OS, IDE, Terminal, or during startup e.g `ASPNETCORE_ENVIRONMENT=Development dotnet run` isn't complicated but it does add another step that we're enforcing on any project that consumes the `SFA.DAS.Configuration` package.

* I agree that it should throw but perhaps not unless it's exhausted all of it's options.
* I run Azurite in a container and it works fine with `UseDevelopmentStorage=true`. I actually prefer it that way as it means less custom config on my machine just for an individual project I may be working on.
* What do you think to at least trying to get it from `appsettings.Development.json` as a fallback?

---

Ah, I'd assumed `UseDevelopmentStorage=true` wouldn't work if Azurite was running in a container!

I've changed it so that on a developer's machine, if the storage connection string environment variable is missing, it defaults to the emulator (to make the developer experience as simple as possible), and if it's running in an actual environment, it fails fast and throws an exception.

(It could be argued though, that if the developer has just run the solution and not set up the config as per the readme, it would be better to throw an exception with the details of what's gone wrong and what they have to do to fix it, rather than default to the emulator, and then they have to figure out why some random class can't be constructed, which is what usually happens when config is missing, as the DI container can't retrieve the config during constructor injection.)